src/index.js

```diff
@@ -4,15 +4,55 @@ import rangeParser from 'parse-numeric-range';
 import shiki from 'shiki';
 import sanitizeHtml from 'sanitize-html';
 
-let highlighter = null;
+// To make sure we only have one highlighter per theme in a process
+const highlighterCache = new Map();
+
+function getThemesFromSettings(settings) {
+  if (!settings.theme)
+    throw new Error(
+      'MDX Pretty Code: no theme was specified in shikiOptions.theme'
+    );
+  if (
+    typeof settings.theme === 'string' ||
+    settings.theme?.hasOwnProperty('tokenColors')
+  ) {
+    return {default: settings.theme};
```

Comment:
I think internally giving single themes the `default` key simplifies things. You could alternatively force people to use the keyed theme format even for single themes.

Reply:
Normalizing internally is fine yep!